### PR TITLE
Implement clean method 

### PR DESCRIPTION
**Pull request by YoonhoKim0527**
implement clean method in distributedrepository

### CODE REVIEW

File: repository/src/lib.rs [hunk 1]
```diff
@@ -96,8 +97,33 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+                let branch_commit = self.raw.locate_branch(branch.to_string().clone()).await?;
+
+                if finalized_branch_commit_hash
+                    != self
+                        .raw
+                        .find_merge_base(branch_commit, finalized_branch_commit_hash)
+                        .await?
+                {
+                    self.raw.delete_branch(branch.to_string()).await?;
+                } else {
```
Comment (junha1) [inline repository/src/lib.rs:121]: Delete this empty else

File: repository/src/lib.rs [hunk 2]
```diff
@@ -96,8 +97,33 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+                }
+            } else {
```
Comment (junha1) [inline repository/src/lib.rs:123]: ditto.

File: repository/src/lib.rs [hunk 3]
```diff
@@ -96,8 +97,33 @@ impl<T: RawRepository> DistributedRepository<T> {
...
-        unimplemented!()
+    pub async fn clean(&mut self) -> Result<(), Error> {
+        let finalized_branch_commit_hash =
+            self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+
+        let branches = self.raw.list_branches().await?;
+
+        for branch in branches {
+            let branch = branch.as_str();
+            if !(branch == WORK_BRANCH_NAME
```
Comment (junha1) [inline repository/src/lib.rs:108]: why not just `branch.as_str() ==` (to save a line, and to reduce the use of `to_string()` down there)

**Review by junha1 — COMMENTED** [review]
You have to remove remote repos too.

**Comment by junha1** [comment]
Please drop the CSV part and introduce it in the later PR.

File: repository/src/lib.rs [hunk 4]
```diff
@@ -96,8 +97,59 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+                        .raw
+                        .find_merge_base(branch_commit, finalized_branch_commit_hash)
+                        .await?
+                {
+                    self.raw.delete_branch(branch.to_string()).await?;
+                }
+            }
+        }
+
+        //remove remote branches
```
Comment (TomTaehoonKim) [inline repository/src/lib.rs:124]:
```suggestion
        // remove remote branches
```

File: repository/src/lib.rs [hunk 5]
```diff
@@ -96,8 +97,59 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+        let remote_list = self.raw.list_remotes().await?;
+        let remote_names: Vec<String> = remote_list.into_iter().map(|(name, _)| name).collect();
+        for remote_name in remote_names {
+            self.raw.remove_remote(remote_name).await?;
+        }
+
+        //check csv
```
Comment (TomTaehoonKim) [inline repository/src/lib.rs:131]:
```suggestion
        // check csv
```

File: repository/src/lib.rs [hunk 6]
```diff
@@ -96,8 +97,59 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+                if finalized_branch_commit_hash
+                    != self
+                        .raw
+                        .find_merge_base(branch_commit, finalized_branch_commit_hash)
+                        .await?
+                {
+                    self.raw.delete_branch(branch.to_string()).await?;
+                }
+            }
+        }
```
Comment (TomTaehoonKim) [inline repository/src/lib.rs:122]: In this loop, outdated `p` branch, `a-#` branches, and `b-#` branches are deleted right?

Can you add a comment before the loop?

File: repository/src/lib.rs (hunk 4 above)
Comment (YoonhoKim0527) [reply repository/src/lib.rs]: fixed.

File: repository/src/lib.rs (hunk 5 above)
Comment (YoonhoKim0527) [reply repository/src/lib.rs]: fixed.

File: repository/src/lib.rs (hunk 6 above)
Comment (YoonhoKim0527) [reply repository/src/lib.rs:123]: added.

File: repository/src/lib.rs [hunk 7]
```diff
@@ -96,8 +97,41 @@ impl<T: RawRepository> DistributedRepository<T> {
...
+
+        // remove remote branches
+        let remote_list = self.raw.list_remotes().await?;
+        let remote_names: Vec<String> = remote_list.into_iter().map(|(name, _)| name).collect();
+        for remote_name in remote_names {
```
Comment (junha1) [inline repository/src/lib.rs:128]: why not just `(remote_name, _) in remote_list`?

Comment (YoonhoKim0527) [reply repository/src/lib.rs]: fixed.

**Review by junha1 — APPROVED** [review]

**Pull request merged**